api/policies/hasValidUsername.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * hasValidUsername
+ *
+ * @module      :: Policy
+ * @description :: Only allows requests that contain a 'username' parameter, which is a string and does not contain spaces
+ * @docs        :: http://sailsjs.org/#!/documentation/concepts/Policies
+ *
+ */
+module.exports = function (req, res, next) {
+  if (req.body.hasOwnProperty('username')) {
+    const username = req.body.username;
+    if (typeof (username) === 'string') {
+      if (username.match(/^[0-9a-zA-Z.@]+$/)){ // . and @ are included to allow for emails
```

Comment:
Can you run 

```
select string_agg(c,'' order by c)
from (
  select distinct regexp_split_to_table(lower(email),'') as c
  from users
) t
```

just to get a list of the unique characters in people's email addresses so they can be added here?  Just realized there might be `_`, for instance

Reply:
-.@0123456789abcdefghijklmnopqrstuvwxyz

Reply:
What does this regex actually check?

Reply:
It checks that there's at least one valid character (alphanumeric, `.`, `-`, or `@`).  Unless you think that usernames should be any unicode?